src/schema/term.rs

```diff
@@ -156,18 +172,13 @@ impl Term {
         self.set_bytes(text.as_bytes());
     }
 
-    /// Removes the value_bytes and set the type code.
-    pub fn clear_with_type(&mut self, typ: Type) {
-        self.truncate(5);
-        self.0[4] = typ.to_code();
-    }
-
-    /// Truncate the term right after the field and the type code.
+    /// Truncates the term. The new length needs to be at least 5, which is reserved for metadata.
     pub fn truncate(&mut self, len: usize) {
```

Comment:
You changed the logic to make sure that the metadata being encoded in same buffer is an implementation detail, and change the semantics of length for `is_empty` , `with_capacity`... But not this one.

Reply:
I saw this relative late and there were some weird dependencies on the internal state, so I simply added an assert.
I fixed it now